twitter/twitter.php

```diff
@@ -680,7 +680,7 @@ function (array $attributes, array $author_contact, $content, $is_quote_share) {
 		}
 
 		// Add the link to the body if the type isn't a photo or there are more than 4 images in the post
-		if (!empty($msgarr['url']) && (($msgarr['type'] != 'photo') || empty($msgarr['images']) || (count($msgarr['images']) > 4))) {
+		if ((strpos($msg, $msgarr['url']) === false) && !empty($msgarr['url']) && (($msgarr['type'] != 'photo') || empty($msgarr['images']) || (count($msgarr['images']) > 4))) {
```

Comment:
I would check if `$msgarr['url']` first before trying to find the URL in the message.
```suggestion
		if (!empty($msgarr['url']) && (strpos($msg, $msgarr['url']) === false) && (($msgarr['type'] != 'photo') || empty($msgarr['images']) || (count($msgarr['images']) > 4))) {
```

Reply:
Details ;-)